### DISCUSSION TITLE
Is there a TCK for verifying implementations?

### DISCUSSION THREAD

**Discussion by bwoester** [VDA5050/VDA5050#238] [Q&A] [answered]
Hello everyone,

I was wondering if there is anything like the [Reactive Streams Technology Compatibility Kit (TCK)](https://github.com/reactive-streams/reactive-streams-jvm/tree/master/tck) for the VDA5050 spec?

I feel like it should help a great deal, both when implementing the spec, as well as when integrating systems from different vendors.

Is there something like it? Maybe using a different name?

**Comment by ConstantinUp** [comment, marked as answer]
Hi @bwoester,
Currently there are only the JSON schemas to check if your implementation of the messages is correct. Within the working group Intralogistics-2X of the VDMA there are plans for something like a VDA 5050 checker. This will be a checker to test if your implementation follows the recommendation. 
But there is no concrete schedule for this project. If you are interested, you can contact the person in charge: [Intralogistics-2X](https://vdma.org/wg-intralogistics-2x).
